packages/loader/container-loader/src/container.ts

```diff
@@ -717,7 +717,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         return this.protocolHandler.quorum;
     }
 
-    public close(error?: ICriticalContainerError) {
+    public close(error?: ICriticalContainerError, reason?: string) {
```

Comment:
I was thinking this over... why not make this required?

Reply:
We can, I was trying to make it less invasive for hosts (i.e. putting myself on their place, I'm not sure what I'd put for reason if I simply want to close it because I no longer need container). Let me know what you think

Reply:
If you don't make it required, let's publicize this well, there could be interesting different cases, like Teams discarding a draft v. typical session close, or something (making this up).  Downside to host's setting this is that it could result in some "noise" in terms of different hosts using different wording for the same logical thing.  But that's true regardless of whether it's required.

All that is to say... if we think this is valuable in our own code paths, I'd say make it required, and see what hosts put. Rather than have it haphazardly filled out by only some hosts/paths.

Also, you can make sure you don't miss any places in our own code this way.  (Although `closeFn` in runtime is harder to update)

Reply:
That said, your PR so you are welcome to leave it, no real harm starting it out optional :)